_posts/en/newsletters/2024-05-01-newsletter.md

```diff
@@ -0,0 +1,243 @@
+lang: en
+---
+This week's newsletter FIXME:harding
+
+## News
+
+- **CTV-like exploding keys proposal:** Tadge Dryja [posted][dryja
+  exploding] to Delving Bitcoin a proposal for a slightly more efficient
+  version of the core idea of [OP_CHECKTEMPLATEVERIFY][topic
+  op_checktemplateverify] (CTV).  With CTV, Alice can pay to an output
+  like:
+
+  ```text
+  OP_CTV <hash>
+  ```
+
+  The preimage for the hash digest is a commitment to the key parts of a
+  transaction, most especially the amount of each output and the script
+  to pay for each output. For example:
+
+  ```text
+  hash(
+    2 BTC to KeyB,
+    3 BTC to KeyC,
+    4 BTC to KeyD
+  )
+  ```
+
+  The `OP_CTV` opcode will succeed if it's executed in a transaction
+  that exactly matches those parameters.  That means Alice's output in
+  one transaction can be spent in a second transaction without requiring
+  an additional signature or any other data, as long as that second
+  transaction matches what Alice expected.
+
+  Dryja suggests an alternative method.  Alice pays to a public key
+  (similar to a [taproot][topic taproot] output but with a different
+  segwit version).  The public key is constructed from a [MuSig2][topic
+  musig] aggregation of one or more actual public keys plus a tweak for
+  each key that securely commits to an amount.  For example (adapted
+  from Dryja's post):
+
+  ```text
+  musig2(
+    KeyB + hash(2 BTC, KeyB)*G,
+    KeyC + hash(3 BTC, KeyC)*G,
+    KeyD + hash(4 BTC, KeyD)*G
+  )
+  ```
+
+  A transaction will be valid if it pays the underlying public keys in
+  exactly the amounts specified.  No signature is required in that case.
+  This saves some space when compared to CTV in taproot, a
+  minimum of about 16 vbytes<!-- for CTV, the internal key needs to be
+  revealed (~8 vbytes) and the CTV tapscript needs to be revealed (~8
+  vbytes)-->.  It seems to use about the same amount of space when compared to CTV
+  in a bare script (i.e., directly placed in an output script).
+
+  When CTV is used in taproot, a keypath spend mutually agreed upon by
+  all involved can be provided as an alternative to executing the CTV,
+  allowing the parties to change where the funds go.  Exploding keys
+  allows the same thing by the people who control KeyB, KeyC, KeyD.
+  The efficiency is the same either way.
+
+  Dryja writes that exploding keys "gives the basic functionality of
+  OP_CTV, while saving a few bytes of witness data. On its own it's
+  maybe not all that compelling, but I wanted to put it out there as
+  maybe it could be a useful primitive as part of a more complex
+  covenant construction."
+
+- **Analyzing a contract protocol with Alloy:** Dmitry Petukhov
+  [posted][petukhov alloy] to Delving Bitcoin a [specification][petukhov
+  spec] he had created using the [Alloy][] specification language for
+  the simple `OP_CAT`-based vault described in [Newsletter #291][news291
+  catvault].  Petukhov used Alloy to [find][petukhov mods] several
```

Comment:
I would add after "... several useful modifications": "and to highlight important constraints to observe for implementors". 

I would add these words because the fact that "envault transaction must not have two outputs" was not apparent, and I actually only understood that after I've made the spec and started model-checking it with Alloy. It showed me these states that I did not expect. It was also something that was not directly related to the covenant, but to the software that implements the whole contract, and that is also a display of usefulness of modelling with Alloy